### PR TITLE
API consistency review updates

### PR DESCRIPTION
**Pull request by srnagar**
This PR updates Azure Monitor Query APIs to address some arch board feedback and to be consistent across other languages.

Fixes https://github.com/Azure/azure-sdk-for-java/issues/22547

### CODE REVIEW

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java [hunk 1]
```diff
@@ -73,6 +75,25 @@ public LogsQueryOptions setIncludeStatistics(boolean includeStatistics) {
         return this;
     }
 
+    /**
+     * If a query has partial errors, the result is returned instead of throwing an exception if this is set to true.
+     * The partial error information is available as part of the query result.
+     *
+     * @return Returns true if partial errors should not throw exception.
+     */
+    public boolean isDisableExceptionOnPartialErrors() {
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java:84]: This seems like an odd name. I understand we try to use is and set for booleans but somehow hasDisabledExceptionOnPartialErrors has a better ring to it.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/MonitorQueryTimeInterval.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        MonitorQueryTimeInterval that = (MonitorQueryTimeInterval) o;
+
+        return Objects.equals(this.duration, that.duration)
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/MonitorQueryTimeInterval.java:214]: Would this not rely on equals override on individual types like duration etc. Do we want to rely on that or explicitly set equality to be value equals?

Comment (pallavit) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/MonitorQueryTimeInterval.java:214]: Ugh - I think I misunderstood this one.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java (hunk 1 above)
Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java]: After discussion with the v-team, renamed the property name to allowPartialErrors and we generally use `is` prefix for boolean getters just to be consistent.

**Review by pallavit — APPROVED** [review]

**Pull request merged**